panos_upgrade_assurance/firewall_proxy.py

```diff
@@ -1094,6 +1121,61 @@ def get_certificates(self) -> dict:
 
         return result
 
+    def get_update_schedules(self) -> dict:
+        """Get schedules for all dynamic updates.
+
+        This method gets scheduled dynamic updates on a device. This includes the ones pushed from Panorama,
+        but it does not include the ones configured via `Panorama/Device Deployment/Dynamic Updates/Schedules`.
+
+        The actual XMLAPI command run here is `config/get` with XPATH set to
+        `/config/devices/entry[@name='localhost.localdomain']/deviceconfig/system/update-schedule`.
+
+        # Returns
+
+        dict: All dynamic updates schedules, key is the entity type to update, like: threats, wildfire, etc.
+
+        ```python showLineNumbers title="Sample output, showing values coming from Panorama"
+        {'@ptpl': 'lab',
```

Comment:
that's a sample output in XML:
```xml
<response status="success" code="19">
  <result total-count="1" count="1">
    <update-schedule ptpl="interfaces" src="tpl">
      <threats ptpl="interfaces" src="tpl">
        <recurring ptpl="interfaces" src="tpl">
          <sync-to-peer ptpl="interfaces" src="tpl">yes</sync-to-peer>
          <daily ptpl="interfaces" src="tpl">
            <at ptpl="interfaces" src="tpl">15:30</at>
            <action ptpl="interfaces" src="tpl">download-and-install</action>
          </daily>
        </recurring>
      </threats>
      <anti-virus ptpl="interfaces" src="tpl">
        <recurring ptpl="interfaces" src="tpl">
          <daily ptpl="interfaces" src="tpl">
            <at ptpl="interfaces" src="tpl">07:45</at>
            <action ptpl="interfaces" src="tpl">download-and-install</action>
          </daily>
        </recurring>
      </anti-virus>
      <global-protect-clientless-vpn ptpl="interfaces" src="tpl">
        <recurring ptpl="interfaces" src="tpl">
          <hourly ptpl="interfaces" src="tpl">
            <at ptpl="interfaces" src="tpl">0</at>
            <action ptpl="interfaces" src="tpl">download-and-install</action>
          </hourly>
        </recurring>
      </global-protect-clientless-vpn>
      <global-protect-datafile ptpl="interfaces" src="tpl">
        <recurring ptpl="interfaces" src="tpl">
          <weekly ptpl="interfaces" src="tpl">
            <at ptpl="interfaces" src="tpl">00:45</at>
            <day-of-week ptpl="interfaces" src="tpl">sunday</day-of-week>
            <action ptpl="interfaces" src="tpl">download-and-install</action>
          </weekly>
        </recurring>
      </global-protect-datafile>
      <wf-private ptpl="interfaces" src="tpl">
        <recurring ptpl="interfaces" src="tpl">
          <every-5-mins ptpl="interfaces" src="tpl">
            <at ptpl="interfaces" src="tpl">1</at>
            <action ptpl="interfaces" src="tpl">download-and-install</action>
          </every-5-mins>
          <sync-to-peer ptpl="interfaces" src="tpl">yes</sync-to-peer>
        </recurring>
      </wf-private>
      <wildfire ptpl="interfaces" src="tpl">
        <recurring src="tpl">
          <every-min src="tpl">
            <action src="tpl">download-and-install</action>
          </every-min>
        </recurring>
      </wildfire>
    </update-schedule>
  </result>
</response>
```

I still do not feel cormortable in doind XML. I do prefer dicts. But if you feel it would be more elegant, you can try to refactor that code.